File: js/data_structures/queue.ts
class Queue<T> {
    #items : T[];
    constructor() {
        this.#items = [];
    }

    enqueue(item: T) {
        this.#items.unshift(item);
    }

    dequeue(): T | null {
        if (this.#items.length > 0)
            return this.#items.pop();
        return null
    }

    peek(): T | null {
        if (this.#items.length > 0)
            return this.#items[this.#items.length-1]
        return null
    }

    size(): number {
        return this.#items.length;
    }
}

const item = new Queue();
item.enqueue(2);
item.enqueue(2);
console.log(item.peek());
console.log(item.dequeue());
console.log(item.size());

File: python/data_structures/queue.py
class Queue():
    def __init__(self):
        self.__items = []

    def enqueue(self, item):
        self.__items.insert(0, item)

    def dequeue(self):
        if len(self.__items) > 0:
            return self.__items.pop()
        return None

    def size(self):
        return len(self.__items)

    def peek(self):
        if len(self.__items) > 0:
            return self.__items[-1]
        return None


item = Queue()
item.enqueue(2)
item.enqueue(2)
print(item.peek())
print(item.dequeue())
print(item.size())
